src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/unusedimports/InputUnusedImportsFromStaticMethodRef.java

```diff
@@ -0,0 +1,91 @@
+/*
+UnusedImports
+processJavadoc = (default)true
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.imports.unusedimports;
+
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.Objects;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Collections;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import static java.lang.String.format; // violation 'Unused import - java.lang.String.format.'
+import static java.util.Arrays.sort; // violation 'Unused import - java.util.Arrays.sort.'
+import static java.util.List.of; // violation 'Unused import - java.util.List.of.'
+import static java.util.Collections.emptyMap; // violation 'Unused import - java.util.Collections.emptyMap.'
+
+/**
+ * This {@link Collections::emptyMap} is NOT a valid link,
```

Comment:
Yes we already have example that uses static import as a valid link 

https://github.com/checkstyle/checkstyle/blob/1e489ea0301f0b699ffd01ccf1dc6daab1ddca0e/src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/unusedimports/InputUnusedImportsFromStaticMethodRefInDocsOnly.java#L26

Reply:
I also added an example with a valid link with `@link` tag

Reply:
> `@see Arrays#sort`

But this isn't a static import. `Arrays` is the name of the class, and like qualified method references from your comment above, it is qualifying the class. You could use this same `see` without the static import and only import `Arrays`.

Reply:
to make sure I understand you well you mean example like this  `@see java.util.Arrays#sort`?

Reply:
Usually for `{@link Arrays#sort}` to work, you need to import the class `java.util.Arrays`.

My question was, if we have a static import `java.util.Collections.emptyMap` is it possible to have some kind of link like `{@link emptyMap}` or `{@link #emptyMap}`.

This is coming from the change that says the link is invalid, but if you can't link anything statically imported, then it seems it would be better to directly say static imports can't be used in javadocs.

Reply:
please consider that this issue is concerned with static import when used with **method reference** your above example has no method reference in it So if there is something unexpected it could be a new but valid issue

Reply:
Then please create a 2nd issue along with the 1st issue https://github.com/checkstyle/checkstyle/pull/14548#discussion_r1530556091

Reply:
> Then please create a 2nd issue along with the 1st issue

@rnveach done

Reply:
> I believe this util is based on regular expression parsing and not Javadoc AST

yes that is why this method in the check is suppressed deprecation it should be moved to AST logic in favor of #11166 

```
    // suppress deprecation until https://github.com/checkstyle/checkstyle/issues/11166
    @SuppressWarnings("deprecation")
    private void collectReferencesFromJavadoc(DetailAST ast) {
        final FileContents contents = getFileContents();
        final int lineNo = ast.getLineNo();
        final TextBlock textBlock = contents.getJavadocBefore(lineNo);
        if (textBlock != null) {
            currentFrame.addReferencedTypes(collectReferencesFromJavadoc(textBlock));
        }
    }

```

> I would move this to the issue it is reported in
I moved all the details to the issue description it is updated now.


> I would want to examine things more before providing feedback.
I am curios to know is the solution for this will be the migration to AST logic. or there can be another solution?

Reply:
@rnveach Also what is the state of this PR now is there something we should do here or we can get this merged

Reply:
@rnveach ping